### PR TITLE
Add rule for incorrect whitespace before comment

### PR DESCRIPTION
**Pull request by LiamPattinson**
Fixes https://github.com/PlasmaFAIR/fortitude/issues/189

We could probably go further and mandate a maximum of 2 spaces too, but this could mess up deliberately lined up comments:

```f90
integer :: a    ! comment
integer :: bb   ! comment
integer :: ccc  ! comment
```

### CODE REVIEW

File: fortitude/src/rules/mod.rs [hunk 1]
```diff
@@ -82,6 +82,7 @@ pub fn code_to_rule(category: Category, code: &str) -> Option<(RuleGroup, Rule)>
         (Style, "061") => (RuleGroup::Stable, Ast, style::end_statements::UnnamedEndStatement),
         (Style, "071") => (RuleGroup::Preview, Ast, style::double_colon_in_decl::MissingDoubleColon),
         (Style, "101") => (RuleGroup::Stable, Text, style::whitespace::TrailingWhitespace),
+        (Style, "121") => (RuleGroup::Preview, Ast, style::whitespace::IncorrectSpaceBeforeComment),
```
Comment (ZedThree) [inline fortitude/src/rules/mod.rs:85]: We're not very consistent with our rule codes, but should this be 111?

File: fortitude/resources/test/fixtures/style/S121.f90 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+! This is fine
+ ! This isn't but we'll let it pass
```
Comment (ZedThree) [inline fortitude/resources/test/fixtures/style/S121.f90:2]: Yeah, catching this will rely on know the indentation of the following node(s)

**Review by ZedThree — APPROVED** [review]

File: fortitude/src/rules/mod.rs (hunk 1 above)
Comment (LiamPattinson) [reply fortitude/src/rules/mod.rs:85]: I'm not sure. I honestly just took a guess that there might be 20-ish whitespace rules we might want to add before this one. Maybe just 102 would make more sense?

Comment (ZedThree) [reply fortitude/src/rules/mod.rs:85]: Yeah, I better there's a few whitespace rules! Let's for for 102 then

Comment (LiamPattinson) [reply fortitude/src/rules/mod.rs]: Fixed in https://github.com/PlasmaFAIR/fortitude/pull/194/commits/ad6455db1917dc15fc752f00757a1ea9e83a8f73

**Pull request merged**